<launch>
    <!-- 启用 use_sim_time 参数 -->
    <param name="/use_sim_time" value="true" />
    <!-- <arg name="project" default="disco_slam"/>      项目名称，默认为disco_slam -->
    <!-- 定义默认值 -->
    <arg name="raw0_cloud" default="/xcj_lidar" />
    <arg name="raw1_cloud" default="/xianfeng_lidar" />
    <arg name="raw2_cloud" default="/gensui_lidar" />
    <arg name="raw0_imu" default="/xcj_imu" />
    <arg name="raw1_imu" default="/xianfeng_imu" />
    <arg name="raw2_imu" default="/gensui_imu" />
    <!-- 这里是代码需要的话题名 -->
    <arg name="input0_cloud" default="jackal0/points_raw" />
    <arg name="input1_cloud" default="jackal1/points_raw" />
    <arg name="input2_cloud" default="jackal2/points_raw" />
    <arg name="input0_imu" default="jackal0/imu_raw" />
    <arg name="input1_imu" default="jackal1/imu_raw" />
    <arg name="input2_imu" default="jackal2/imu_raw" />
    <arg name="correct0_imu" default="jackal0/imu_correct" />
    <arg name="correct1_imu" default="jackal1/imu_correct" />
    <arg name="correct2_imu" default="jackal2/imu_correct" />

    <!-- 启动 rosbag play 节点 -->
    <!-- <node name="rosbag_player" pkg="rosbag" type="play"  -->
    <!-- args="$(env HOME)/rosbag/disco/kitti08_2.bag -l -/-clock"/> -->

    <!-- 定义参数 -->
    <arg name="no"      default="1"/>  <!-- 编号参数 -->
    <arg name="project" default="disco_slam"/>  <!-- 项目名称 -->
    <arg name="robot0"	default="jackal0"/>  <!-- 机器人0名称 -->
    <arg name="robot1"	default="jackal1"/>  <!-- 机器人1名称 -->
    <arg name="robot2"	default="jackal2"/>  <!-- 机器人2名称 -->

    <!-- 加载参数文件 -->
    <!-- PARK -->
    <!-- <rosparam file="$(find disco_slam)/config/params.yaml" command="load" /> -->
    <!-- <rosparam file="$(find disco_slam)/src/DiSCo-SLAM/config/mapfusion.yaml" command="load" /> -->
    <!-- KITTI-08 -->
    <!-- <rosparam file="$(find disco_slam)/config/params_k.yaml" command="load" />
    <rosparam file="$(find disco_slam)/src/DiSCo-SLAM/config/mapfusion_k.yaml" command="load"/> -->
    <!-- UE5-1 -->
    <rosparam file="$(find disco_slam)/config/params_ue_1.yaml" command="load" />
    <rosparam file="$(find disco_slam)/src/DiSCo-SLAM/config/mapfusion_ue_1.yaml" command="load"/>
    <!-- 启动机器人0的LOAM模块 -->
    <include file="$(find disco_slam)/launch/include/module_loam.launch" >
        <arg name="robot" value="$(arg robot0)"/>
        <arg name="id1" value="$(arg robot1)"/>
        <arg name="id2" value="$(arg robot2)"/>
        <arg name="no" value="$(arg no)"/>
    </include>

    <!-- 启动机器人1的LOAM模块 -->
    <include file="$(find disco_slam)/launch/include/module_loam.launch" >
        <arg name="robot" value="$(arg robot1)"/>
        <arg name="id1" value="$(arg robot0)"/>
        <arg name="id2" value="$(arg robot2)"/>
        <arg name="no" value="$(arg no)"/>
    </include>

    <!-- 启动机器人2的LOAM模块 -->
    <include file="$(find disco_slam)/launch/include/module_loam.launch" >
        <arg name="robot" value="$(arg robot2)"/>
        <arg name="id1" value="$(arg robot0)"/>
        <arg name="id2" value="$(arg robot1)"/>
        <arg name="no" value="$(arg no)"/>
      </include>

    <!-- 机器人状态TF转换 (已注释) -->
    <!-- <include file="$(find disco_slam)/launch/include/module_robot_state_publisher.launch" /> -->

    <!-- 运行导航卫星模块 (已注释) -->
    <!-- <include file="$(find disco_slam)/launch/include/module_navsat.launch" /> -->

    <!-- 运行Rviz可视化工具 -->
    <include file="$(find disco_slam)/launch/include/module_rviz_ue_1.launch" />

    <!-- 运行通信模拟器 (已注释) -->
    <!-- <node pkg="$(arg project)" name= "$(arg project)_listener" type="listener.py" output="screen"/> -->

    <!-- /mnt/c/Users/Greary/Documents/rosbag/lidar_1_ring_06_03.bag -->
    <!-- /mnt/d/jkg_rosbag/E2-disco/kitti08_2.bag -->
    <node pkg="rosbag" type="play" name="bag_player"
        args=" /mnt/c/Users/Greary/Documents/rosbag/lidar_1_ring_06_03.bag
        -l --clock -r 1">
        <!-- 话题重映射 -->
        <!-- <remap from="原始话题名称" to="新话题名称" /> -->
        <remap from="$(arg raw0_cloud)" to="$(arg input0_cloud)" />
        <remap from="$(arg raw1_cloud)" to="$(arg input1_cloud)" />
        <remap from="$(arg raw2_cloud)" to="$(arg input2_cloud)" />
        <remap from="$(arg raw0_imu)" to="$(arg input0_imu)" />
        <remap from="$(arg raw1_imu)" to="$(arg input1_imu)" />
        <remap from="$(arg raw2_imu)" to="$(arg input2_imu)" />
        <remap from="$(arg raw0_imu)" to="$(arg correct0_imu)" />
        <remap from="$(arg raw1_imu)" to="$(arg correct1_imu)" />
        <remap from="$(arg raw2_imu)" to="$(arg correct2_imu)" />
    </node>
    

</launch>


